server.go

```diff
@@ -182,8 +183,11 @@ func (s *Server) handleNextGame(rw http.ResponseWriter, req *http.Request) {
 		sort.Strings(words)
 	}
 
-	g := newGame(request.GameID, randomState(words))
-	s.games[request.GameID] = g
+	g, ok := s.games[request.GameID]
+	if (!ok || request.CreateNew) {
+		g := newGame(request.GameID, randomState(words))
```

Comment:
```suggestion
		g = newGame(request.GameID, randomState(words))
```

In Go, `:=` defines new a new variable, rather than writing to an existing. This will create a new variable in the new scope of the if branch, shadowing the outer scope's `g` variable. The `writeGame` function would then write the old shadowed `g` value, rather than the new game that was just created and saved to the games map.

Changing `:=` to `=` will write to that outer scope's `g` and ensure that we set and write the same Game struct.